### PR TITLE
Treat Mamba as preferred package manager in embedded documentation

### PR DESCRIPTION
**Pull request by mattwthompson**
#1706 

- [x] Tag issue being addressed
- [ ] Add [tests](https://github.com/openforcefield/openff-toolkit/tree/main/openff/toolkit/_tests)
- [ ] Update docstrings/[documentation](https://github.com/openforcefield/openff-toolkit/tree/main/docs), if applicable
- [ ] [Lint](https://open-forcefield-toolkit.readthedocs.io/en/latest/developing.html#style-guide) codebase
- [ ] Update [changelog](https://github.com/openforcefield/openff-toolkit/blob/main/docs/releasehistory.rst)

### CODE REVIEW

File: docs/users/developing.md [hunk 1]
```diff
@@ -335,9 +335,10 @@ These checks will use the most recent versions of each linter.
 To ensure that changes follow these standards, you can install and run these tools locally:
 
 ```shell
-conda install black isort -c conda-forge
+mamba install black isort flake8 -c conda-forge
```
Comment (j-wags) [inline docs/users/developing.md:338]: (not blocking) Do we also need flake8-bugbear?

**Review by j-wags — APPROVED** [review]
Looks great - One proposed change I feel somewhat strongly about. Once that's handled, feel free to merge!

File: FAQ.md [hunk 2]
```diff
@@ -70,19 +70,19 @@ As of August 2022, some upstreams (at least AmberTools, possibly more) are not b
...
-conda create -n <my_new_env> -c conda-forge openff-toolkit
-conda activate <my_new_env>
+mamba create -n <my_new_env> -c conda-forge openff-toolkit
+mamba activate <my_new_env>
 ```
 
-Installing into a new environment avoids forcing conda to satisfy the dependencies of both the toolkit and all existing packages in that environment.
-Taking the approach that conda environments are generally disposable, even ephemeral, minimizes the chances for hard-to-diagnose dependency issues.
+Installing into a new environment avoids forcing mamba to satisfy the dependencies of both the toolkit and all existing packages in that environment.
+Taking the approach that virtual environments are generally disposable, even ephemeral, minimizes the chances for hard-to-diagnose dependency issues.
```
[the remark this answers is not in the document]
Comment (mattwthompson) [reply FAQ.md:83]: "virtual environment" is the umbrella term for all solutions in this space, of which conda environments are subsets - I don't think it's synonymous with `venv`, `virtualenv`, `pyenv`, or any similar tools

File: docs/users/developing.md (hunk 1 above)
Comment (mattwthompson) [reply docs/users/developing.md:338]: Strictly speaking, plugins should be installed here, but I don't think it's too important and I'd rather keep the pre-commit config as the sole source of truth. With the toolkit's config, there shouldn't be many cases in which the base package passes but a plugin throws an error

**Pull request merged**